### ISSUE TITLE
Poor ambient occlusion performance

### ISSUE THREAD

**Issue by makamys** [FalsePattern/Triangulator#26]
If smooth lighting is enabled, the mod reduces the framerate while chunks are being updated. This happens even with all options in the config disabled. When smooth lighting is turned off however, the mod doesn't affect framerate.

Comparing profiler snapshots, I have found `renderStandardBlockWithAmbientOcclusion` to be the culprit: its CPU time increases from 10% to 25% with Triangulator. So the issue seems to be caused by @embeddedt's smooth lighting fix.

## Frametime graphs (chunk updates begin in the middle)
Without Triangulator (~130 fps):
![](https://user-images.githubusercontent.com/13667520/181380839-92c1c002-e5d6-490c-800f-cd409fdd7d2a.png)
With Triangulator (~100 fps):
![](https://user-images.githubusercontent.com/13667520/181380852-564d9669-e6a6-4f51-8547-0e8df883890a.png)

## Snapshots (.nps)
[snapshots.zip](https://github.com/FalsePattern/Triangulator/files/9203916/snapshots.zip)

## Mods
```
falsepatternlib-mc1.7.10-0.9.4.jar
mixingasm-1.7.10-0.2.1.jar
sampler-1.60.1.jar
SpongeMixins-1.4.0.jar
triangulator-mc1.7.10-1.7.3.jar
```

**Comment by embeddedt** [comment]
This might be due to the Inject allocating a CallbackInfo object every time. Maybe the Redirect version pushed recently to master would fix that?

**Comment by makamys** [comment]
I'm using 1.7.3, which has the latest commit.

**Comment by FalsePattern** [comment]
i am currently working on a full rewrite of the block rendering system inside the https://github.com/FalsePattern/Triangulator/tree/rewrite branch, replacing vanilla code, using a full overwrite, with improved and more modular render code

**Comment by FalsePattern** [comment]
Should be fixed in 2.0 alpha 1, i made the mixin for the target an `@Overwite` instead, which has zero overhead (with a fallback config option for enabling `@Inject` mode in case another mod also mixes into those 2 methods.

**Comment by makamys** [comment]
Nice! I repeated the experiment with 2.0, and the issue appears to be gone.

Without Triangulator (~133 fps)
![](https://user-images.githubusercontent.com/13667520/191013650-161f296d-bac6-46d6-98c0-2d437729c028.png)

With Triangulator 2.0 (~134 fps)
![](https://user-images.githubusercontent.com/13667520/191014191-8eff2543-70dd-483d-b9b8-b6d557aafc25.png)

With Triangulator 1.7.3 (~111 fps)
![](https://user-images.githubusercontent.com/13667520/191013808-cefe6bfa-edfb-442e-bc3a-fcc049989c69.png)

## Mods
```
00gasstation-mc1.7.10-0.3.2.jar
falsepatternlib-mc1.7.10-0.10.10.jar
sampler-1.60.1.jar
[nothing] OR triangulator-mc1.7.10-2.0.0-alpha0001.jar OR triangulator-mc1.7.10-1.7.3.jar
```
